Realm/RLMQueryUtil.mm

```diff
@@ -636,13 +664,13 @@ void add_collection_operation_constraint(NSPredicateOperatorType operatorType,
     if (diacriticSensitive) {
         switch (operatorType) {
             case NSBeginsWithPredicateOperatorType:
-                m_query.and_query(column.begins_with(value, caseSensitive));
+                add_substring_constraint(std::forward<T>(value), column.begins_with(value, caseSensitive));
```

Comment:
You want `std::move` rather than `std::forward` here, since `value` isn't a forwarding reference.